### PR TITLE
refactor: Unicode with curly braces

### PR DESCRIPTION
**Pull request by vanillajonathan**
This takes off where #3571 and #3569 left off, and changes the syntax for the Unicode escape sequence to use curly braces because maybe you wanted that instead?

### CODE REVIEW

File: crates/prql-parser/src/lexer.rs [hunk 1]
```diff
@@ -347,29 +347,19 @@ fn escaped_character() -> impl Parser<char, char, Error = Cheap<char>> {
         just('n').to('\n'),
         just('r').to('\r'),
         just('t').to('\t'),
-        (just('u').ignore_then(
+        (just("u{").ignore_then(
```
Comment (max-sixty) [inline crates/prql-parser/src/lexer.rs:350]: The canonical way would be to use `.delimited_by`, but this is OK given it seems to work and these can't be nested

File: CHANGELOG.md [hunk 2]
```diff
@@ -40,11 +40,7 @@
   written in Go and used by the static website generator
   [Hugo](https://gohugo.io/). (@vanillajonathan, #3597)
 - [scc](https://github.com/boyter/scc), a source lines of code counter now has
-  support for `.prql` files. (@vanillajonathan)
-- [gcloc](https://github.com/JoaoDanielRufino/gcloc) a source lines of code
-  counter now has support for `.prql` files. (@vanillajonathan)
-- [cloc](https://github.com/AlDanial/cloc) a source lines of code counter now
-  has support for `.prql` files. (@AlDanial)
```
Comment (max-sixty) [inline CHANGELOG.md]: Did we intend to remove these?

**Review by max-sixty — APPROVED** [review]
Awesome! Thanks @vanillajonathan !

**Comment by max-sixty** [comment]
For the tests to pass, we need to remove the `no-fmt` in the example at `reference/syntax/strings/quoting-and-escape-characters`, since the formatting now works...

File: CHANGELOG.md (hunk 2 above)
Comment (vanillajonathan) [reply CHANGELOG.md]: No, we do not intend to remove those. Good spotted, my mistake.

**Comment by vanillajonathan** [comment]
> For the tests to pass, we need to remove the `no-fmt` in the example at `reference/syntax/strings/quoting-and-escape-characters`, since the formatting now works...

Thanks!

File: crates/prql-parser/src/lexer.rs (hunk 1 above)
Comment (vanillajonathan) [reply crates/prql-parser/src/lexer.rs:350]: Oh, I didn't know about that one. We could use that one instead if you want, but then we can't use `at_least` and `at_most`. I noticed that the Rust compiler is friendly enough to actually tell you that you can at most use 6 characters if you try to use more than that.

Comment (max-sixty) [reply crates/prql-parser/src/lexer.rs:350]: I think my message was unclear. We can still do the `at_least` / `at_most`. The `.delimited_by` would be for the brackets. Here's an example of it: https://github.com/PRQL/prql/blob/33b71715136380255482e902c8afe92bd5a8fe10/crates/prql-parser/src/expr.rs#L114

File: CHANGELOG.md [hunk 3]
```diff
@@ -40,7 +40,7 @@
   written in Go and used by the static website generator
   [Hugo](https://gohugo.io/). (@vanillajonathan, #3597)
 - [scc](https://github.com/boyter/scc), a source lines of code counter now has
-  support for `.prql` files. (@vanillajonathan)
+  support for `.prql` files.
```
Comment (max-sixty) [inline CHANGELOG.md:43]: Did we intend to remove this?

**Comment by max-sixty** [comment]
This is great, so I'll merge, but fine to do any small refinements as a follow up...

**Comment by max-sixty** [comment]
One small follow-up that would be nice is to be explicit that we've changed the syntax from `uXXXX` to `u{XXXXX}` (better expressed, hopefully!)

File: CHANGELOG.md (hunk 3 above)
Comment (vanillajonathan) [reply CHANGELOG.md:43]: No. Thanks for spotting.

**Pull request merged**